### DISCUSSION TITLE
Cast button not working and root keep crashing

### DISCUSSION THREAD

**Discussion by noweir** [ReVanced/revanced-discussions#389] [Question] [answered]
I exclude hide cast button and even done exclusive same error so cast button is there any fix or i need to wait more

And about build without micro keep crash every time i open it android 12

**Comment by TheJeterLP** [comment, marked as answer]
Casting only works on non-root version. Hiding the cast button is a MUST for the non-root version since casting is not working with microg.

**Reply by ghost** [reply]
i still didnt modify the microg patch in a way that only castv1 works :D

**Reply by TheJeterLP** [reply]
yea, but microg needs to fix castv2

**Reply by TheJeterLP** [reply]
hide-cast-button is a dependency of the microg patch

**Reply by ghost** [reply]
which they probably wont

**Reply by TheJeterLP** [reply]
we still cannot fix that

**Reply by ghost** [reply]
> hide-cast-button is a dependency of the microg patch

it can be modified so only castv1 works

play services have the ability to fall back to v1 if v2 isnt supported

**Reply by TheJeterLP** [reply]
That would require a creation of our own microg then. I don't know if this is desired

**Reply by ghost** [reply]
modifying microg patch should be enough
no need to fork microg

**Reply by TheJeterLP** [reply]
If thats the case, would you create an Issue for this? And if you want, maybe have alook into this feature?

**Reply by ghost** [reply]
sure
i did try to only disable only castv2 before but it broke the button lol
